boa_engine/src/bytecompiler/declarations.rs

```diff
@@ -1055,15 +1086,27 @@ impl ByteCompiler<'_, '_> {
                     else {
                         // a. Let initialValue be ! env.GetBindingValue(n, false).
                         let binding = self.get_binding_value(n);
-                        let index = self.get_or_insert_binding(binding);
-                        self.emit(Opcode::GetName, &[Operand::U32(index)]);
+                        match self.get_or_insert_binding(binding) {
+                            EnvironmentAccess::Fast { index } => {
+                                self.emit(Opcode::GetLocal, &[Operand::U32(index)]);
+                            }
+                            EnvironmentAccess::Slow { index } => {
+                                self.emit(Opcode::GetName, &[Operand::U32(index)]);
+                            }
+                        }
```

Comment:
I'm seeing this snippet several times. Should this be extracted into its own method?

Reply:
Will do this once I have finished the optimization, currently there are some bugs which are making some test test22 fails